data_structures/skip_list.cpp

```diff
@@ -0,0 +1,228 @@
+/**
+ * A skip list is a data structure that is used for storing a sorted list of items with a
+ *  help of hierarchy of linked lists that connect increasingly sparse subsequences of the items
+ * 
+ *  References used: GeeksForGeeks skip list code, https://iq.opengenus.org/skip-list/  Code and PseudoCode.
+*/
+
+#include <iostream>
+#include <vector>
+#include <cstring>
+#include <ctime>   
+
+// using namespace std; 
+using std::vector;
+using std::endl;
+
+#define MAX_LEVEL 2   ///< maximum level of skip list
+#define PROBABILITY 0.5      ///< current probability for "coin toss"
+  
+
+/**
+ *  Node structure [Key][Node*, Node*...]
+*/
+struct Node { 
+    int key;
+    //pointer of value 
```

Comment:
Please add a space after the comment:
```c++
    // Comment (good)
    //Comment (wrong)
```
Same in other parts of the code.

Reply:
ok